src/audio/multiband_drc/multiband_drc.c

```diff
@@ -94,94 +94,88 @@ static int multiband_drc_eq_init_coef_ch(struct sof_eq_iir_biquad *coef,
 	return 0;
 }
 
+/**
+ * @brief Initialize coefficients for multiband DRC processing.
+ *
+ * Allocates and initializes filter coefficients for the multiband DRC module.
+ * Each coefficient array (crossover, emphasis, and de-emphasis) is embedded
+ * and initialized in the component data structure.
+ *
+ * The function checks for configuration validity and adheres to predefined
+ * channel and band count limits.
+ *
+ * The memory layout for the coefficients is optimized for performance.
+ *
+ * @param mod Pointer to the processing module containing multiband DRC data.
+ * @param nch Number of channels to process, up to PLATFORM_MAX_CHANNELS.
+ * @param rate Sampling rate, not used in the current implementation.
+ *
+ * @return 0 on success, or a negative error code on failure (e.g., invalid configuration,
+ *         memory allocation failure).
+ */
 static int multiband_drc_init_coef(struct processing_module *mod, int16_t nch, uint32_t rate)
 {
 	struct comp_dev *dev = mod->dev;
 	struct multiband_drc_comp_data *cd = module_get_private_data(mod);
-	struct sof_eq_iir_biquad *crossover;
-	struct sof_eq_iir_biquad *emphasis;
-	struct sof_eq_iir_biquad *deemphasis;
 	struct sof_multiband_drc_config *config = cd->config;
 	struct multiband_drc_state *state = &cd->state;
 	uint32_t sample_bytes = get_sample_bytes(cd->source_format);
+	struct sof_eq_iir_biquad *crossover, *emphasis, *deemphasis;
 	int i, ch, ret, num_bands;
 
+	/* Check for valid configuration */
 	if (!config) {
 		comp_err(dev, "multiband_drc_init_coef(), no config is set");
 		return -EINVAL;
 	}
 
 	num_bands = config->num_bands;
 
-	/* Sanity checks */
```

Comment:
personally i don't see the value in this deletion, you are deleting a minor amount of instructions to just make the log a less easier to understand as now its "1 of 2 are failing".

Reply:
Sure, I will revert back to initial implementation

Reply:
ping on revertion